diff -Naur cross-sh/build_toolchain.sh cross-sh-mingw/build_toolchain.sh
--- cross-sh/build_toolchain.sh	2022-09-18 13:08:24.964526719 -0400
+++ cross-sh-mingw/build_toolchain.sh	2022-09-18 13:06:51.132172656 -0400
@@ -14,36 +14,14 @@
 echo "Building Binutils..."
 bash toolchain_binutils.sh &> /dev/null
 error_check "$?"
+echo "Building mingw headers"
+bash toolchain_mingw_headers.sh &> /dev/null
 echo "Building GCC Stage 1..."
 bash toolchain_gcc_static.sh &> /dev/null
 error_check "$?"
-echo "Copying the Linux API Headers..."
-bash toolchain_linux_headers.sh &> /dev/null
-error_check "$?"
-if [[ $LIBC == GLIBC ]]
-then
-   echo "Building GLIBC"
-   bash toolchain_glibc.sh &> /dev/null
-   error_check "$?"
-elif [[ $LIBC == MUSL ]]
-then
-   echo "Building Musl"
-   bash toolchain_musl.sh &> /dev/null
-   error_check "$?"
-elif [[ $LIBC == PICOLIBC ]]
-then
-   bash toolchain_picolibc &> /dev/null
-   error_check "$?"
-elif [[ $LIBC == NEWLIB ]]
-then
-    bash toolchain_newlib.sh &> /dev/null
-    error_check "$?"
-else 
-	echo "LIBC Variable is invalid check your toolchain.conf"
-fi
-
-
+echo "Building mingw"
+bash toolchain_mingw.sh &> /dev/null
 echo "Building the final stage of GCC..."
 bash toolchain_gcc_final.sh &> /dev/null
 error_check "$?"
-echo "All Done, remember to add $LOCATION/usr/bin to your path"
+echo "Run package.sh"
diff -Naur cross-sh/package.sh cross-sh-mingw/package.sh
--- cross-sh/package.sh	1969-12-31 19:00:00.000000000 -0500
+++ cross-sh-mingw/package.sh	2022-09-18 13:06:51.133172660 -0400
@@ -0,0 +1,10 @@
+#!/bin/bash
+source toolchain.conf
+mkdir -p /pkgs/$TARGET_CROSS/var/cache/mercury/meta-pkgs
+echo "$TARGET_CROSS-binutils $TARGET_CROSS-gcc $TARGET_CROSS-glibc $TARGET_CROSS-linux-api-headers" > /pkgs/$TARGET_CROSS/depend
+echo "$TARGET_CROSS-binutils $TARGET_CROSS-gcc $TARGET_CROSS-glibc $TARGET_CROSS-linux-api-headers" > /pkgs/$TARGET_CROSS/var/cache/mercury/meta-pkgs/$TARGET_CROSS
+cd /pkgs
+tar -cvzpf $TARGET_CROSS-gcc.tar.xz $TARGET_CROSS-gcc
+tar -cvpzf $TARGET_CROSS-binutils.tar.xz $TARGET_CROSS-binutils
+tar -cvpzf $TARGET_CROSS-mingw.tar.xz $TARGET_CROSS-mingw
+mv $TARGET_CROSS*.tar.xz /finished
diff -Naur cross-sh/toolchain_binutils.sh cross-sh-mingw/toolchain_binutils.sh
--- cross-sh/toolchain_binutils.sh	2022-09-18 13:08:24.964526719 -0400
+++ cross-sh-mingw/toolchain_binutils.sh	2022-09-18 13:06:47.990160800 -0400
@@ -15,12 +15,27 @@
 mkdir build
 cd build
 
-../configure --prefix=/usr --with-sysroot=$LOCATION --disable-nls --disable-werror --target=$TARGET_CROSS
+../configure --prefix=/usr --with-sysroot=$LOCATION/$TARGET_CROSS --disable-nls --disable-werror --target=$TARGET_CROSS
 
 make -j$JOBS
 
 make DESTDIR=$LOCATION install
-
+make DESTDIR=/pkgs/$TARGET_CROSS-$pkg install
 # Delete the old files
 cd $BUILD_DIR
 rm -rf $pkg-$LD_VER
+# Remove conflicting files conflicting with the host
+cd /pkgs/$TARGET_CROSS-$pkg
+rm -rf usr/share/info
+rm -rf usr/include
+rm -rf usr/etc/gprofng.rc
+rm -rf usr/lib/gprofng
+# Strip 
+cd /pkgs/$TARGET_CROSS-$pkg/usr/bin
+for binary in $(ls .)
+do
+	if [[ ! "$binary" == "$TARGET_CROSS-gp-display-html" ]]
+	then
+		strip $binary
+	fi
+done
diff -Naur cross-sh/toolchain.conf cross-sh-mingw/toolchain.conf
--- cross-sh/toolchain.conf	2022-09-18 13:08:24.964526719 -0400
+++ cross-sh-mingw/toolchain.conf	2022-09-18 13:06:51.133172660 -0400
@@ -1,32 +1,30 @@
 #!/bin/bash
 
-# Put your prefences here
+# This is the config for building the Tucana Packages
+# If you want to use this to build for another distro,
+# Check out cross-sh, it is distro agnostic
 
-SOURCES_DIR=$(pwd)/sources
+SOURCES_DIR=/blfs/builds
 START_DIR=$(pwd)
-BUILD_DIR=$(pwd)/build
-TARGET_CROSS=x86_64-linux-musl
-LOCATION=/opt/cross
+BUILD_DIR=/blfs/builds
+TARGET_CROSS=x86_64-w64-mingw32
+LOCATION=/
 HOST=x86_64-linux-gnu
 PATH=$LOCATION/usr/bin:$PATH
-JOBS=20
-# Select your libc here don't forget to set the version below
-# Available options are GLIBC, NEWLIB, MUSL or PICOLIBC  if you want java compatiablity choose glibc otherwise it is reccommended to choose musl
-LIBC=MUSL
+JOBS=$(nproc)
+
+# DO NOT CHANGE, THIS BUILD HAS BEEN PATCHED FOR MINGW ONLY!!!
+LIBC=MINGW
 
 
-# For the kernel, check here for what to put https://github.com/torvalds/linux/tree/master/arch.  Leave it blank for host kernel
-ARCH=x86_64
 
 # Package Versions
-LD_VER=2.38
-CC_VER=12.1.0
-KERNEL_VER=5.18.10
-GLIBC_VER=2.35
+LD_VER=2.39
+CC_VER=12.2.0
 
 # Other options for libc, select which one you want above
 # https://www.musl-libc.org/
-MUSL_VER=1.2.3
+#MUSL_VER=1.2.3
 # https://sourceware.org/newlib/
 #NEWLIB_VERSION
 # https://github.com/picolibc/picolibc
diff -Naur cross-sh/toolchain_gcc_final.sh cross-sh-mingw/toolchain_gcc_final.sh
--- cross-sh/toolchain_gcc_final.sh	2022-09-18 13:08:24.964526719 -0400
+++ cross-sh-mingw/toolchain_gcc_final.sh	2022-09-18 13:08:08.013462756 -0400
@@ -15,21 +15,17 @@
 
 # Start the build
 
-tar -xf $SOURCES_DIR/mpfr-4.1.0.tar.xz
-mv -v mpfr-4.1.0 mpfr
-tar -xf $SOURCES_DIR/gmp-6.2.1.tar.xz
-mv -v gmp-6.2.1 gmp
-tar -xf $SOURCES_DIR/mpc-1.2.1.tar.gz
-mv -v mpc-1.2.1 mpc
 
 mkdir build
 cd build
 
-../configure --prefix=/usr --enable-languages=c,c++ --disable-multilib --target=$TARGET_CROSS --with-sysroot=$LOCATION --disable-werror 
+../configure --prefix=/usr --disable-shared --disable-threads --enable-languages=c,c++ --disable-multilib --target=x86_64-w64-mingw32 --with-build-sysroot=/usr/x86_64-w64-mingw32 --disable-libstdcxx --disable-libatomic
 
-make -j$JOBS
 
-sudo make PATH=$LOCATION/usr/bin:$PATH DESTDIR=$LOCATION install
+make -j$(nproc)
+make DESTDIR=$LOCATION install
+make DESTDIR=/pkgs/$TARGET_CROSS-$pkg install
+find /pkgs/$TARGET_CROSS-$pkg/usr/bin/ -type f -and \( -executable \) -exec strip '{}' \;
 
 # Delete the old files
 cd $BUILD_DIR
diff -Naur cross-sh/toolchain_gcc_static.sh cross-sh-mingw/toolchain_gcc_static.sh
--- cross-sh/toolchain_gcc_static.sh	2022-09-18 13:08:24.964526719 -0400
+++ cross-sh-mingw/toolchain_gcc_static.sh	2022-09-18 13:06:51.134172663 -0400
@@ -2,7 +2,7 @@
 set -e
 # Source the configuration
 source toolchain.conf
-
+# For build system only
 pkg=gcc
 # Untar
 
@@ -12,21 +12,15 @@
 
 # Start the build
 
-tar -xf $SOURCES_DIR/mpfr-4.1.0.tar.xz
-mv -v mpfr-4.1.0 mpfr
-tar -xf $SOURCES_DIR/gmp-6.2.1.tar.xz
-mv -v gmp-6.2.1 gmp
-tar -xf $SOURCES_DIR/mpc-1.2.1.tar.gz
-mv -v mpc-1.2.1 mpc
 
 mkdir build
 cd build
 
-../configure --prefix=/usr --disable-threads --disable-shared --with-newlib --without-headers --enable-languages=c,c++ --disable-multilib --target=$TARGET_CROSS
+../configure --prefix=/usr --disable-threads --disable-shared --enable-languages=c,c++ --disable-multilib --target=$TARGET_CROSS
 
-make all-gcc all-target-libgcc -j$JOBS
+make all-gcc -j$JOBS
 
-PATH=$LOCATION/usr/bin:$PATH make DESTDIR=$LOCATION install-gcc install-target-libgcc
+make install-gcc
 
 # Delete the old files
 cd $BUILD_DIR
diff -Naur cross-sh/toolchain_glibc.sh cross-sh-mingw/toolchain_glibc.sh
--- cross-sh/toolchain_glibc.sh	2022-09-18 13:08:24.964526719 -0400
+++ cross-sh-mingw/toolchain_glibc.sh	1969-12-31 19:00:00.000000000 -0500
@@ -1,26 +0,0 @@
-#!/bin/bash
-set -e
-# Source the configuration
-source toolchain.conf
-
-pkg=glibc
-# Untar
-
-cd $BUILD_DIR
-tar -xvf $SOURCES_DIR/$pkg-$GLIBC_VER.tar.xz
-cd $pkg-$GLIBC_VER
-
-# Start the build
-
-mkdir build
-cd build
-
-../configure --prefix=/usr --host=$TARGET_CROSS --with-headers=$LOCATION/usr/include
-
-make -j$JOBS
-
-make DESTDIR=$LOCATION install
-
-# Delete the old files
-cd $BUILD_DIR
-rm -rf $pkg-$GLIBC_VER
diff -Naur cross-sh/toolchain_linux_headers.sh cross-sh-mingw/toolchain_linux_headers.sh
--- cross-sh/toolchain_linux_headers.sh	2022-09-18 13:08:24.964526719 -0400
+++ cross-sh-mingw/toolchain_linux_headers.sh	1969-12-31 19:00:00.000000000 -0500
@@ -1,25 +0,0 @@
-#!/bin/bash
-
-# These are usually called linux-api-headers on different distros, just in case you have some trouble with these
-
-# Exit on error (not really needed for this package)
-set -e
-# Source the configuration
-source toolchain.conf
-
-# This is what the tar is called
-pkg=linux
-# Untar
-
-cd $BUILD_DIR
-tar -xvf $SOURCES_DIR/$pkg-$KERNEL_VER.tar.xz
-cd $pkg-$KERNEL_VER
-
-# Start the build
-
-make ARCH=$ARCH headers
-cp -rpv usr/include/* $LOCATION/usr/include
-
-# Delete the old files
-cd $BUILD_DIR
-rm -rf $pkg-$KERNEL_VER
diff -Naur cross-sh/toolchain_mingw_headers.sh cross-sh-mingw/toolchain_mingw_headers.sh
--- cross-sh/toolchain_mingw_headers.sh	1969-12-31 19:00:00.000000000 -0500
+++ cross-sh-mingw/toolchain_mingw_headers.sh	2022-09-18 13:06:51.134172663 -0400
@@ -0,0 +1,24 @@
+#!/bin/bash
+set -e
+# Source the configuration
+source toolchain.conf
+# For Build System Only
+pkg=mingw
+# Untar
+
+cd $BUILD_DIR
+tar -xvf $SOURCES_DIR/mingw-w64-v10.0.0.tar.bz2
+cd mingw-w64-v10.0.0
+
+# Start the build
+
+cd mingw-w64-headers
+
+./configure --prefix=/usr/$TARGET_CROSS 
+
+make -j$JOBS
+
+make install
+# Delete the old files
+cd $BUILD_DIR
+rm -rf mingw-w64-v10.0.0
diff -Naur cross-sh/toolchain_mingw.sh cross-sh-mingw/toolchain_mingw.sh
--- cross-sh/toolchain_mingw.sh	1969-12-31 19:00:00.000000000 -0500
+++ cross-sh-mingw/toolchain_mingw.sh	2022-09-18 13:06:51.134172663 -0400
@@ -0,0 +1,34 @@
+#!/bin/bash
+set -e
+# Source the configuration
+source toolchain.conf
+# For Build System Only
+pkg=mingw
+# Untar
+
+cd $BUILD_DIR
+tar -xvf $SOURCES_DIR/mingw-w64-v10.0.0.tar.bz2
+cd mingw-w64-v10.0.0
+
+# Start the build
+
+cd mingw-w64-headers
+
+./configure --prefix=/usr/$TARGET_CROSS 
+
+make
+make DESTDIR=/pkgs/$TARGET_CROSS-mingw install
+make install
+
+cd ..
+
+./configure --enable-experimental --prefix=/usr/x86_64-w64-mingw32 --host=x86_64-w64-mingw32
+# For some reason mingw doesn't support multiple jobs
+make
+
+make DESTDIR=/pkgs/$TARGET_CROSS-mingw install
+make install
+
+# Delete the old files
+cd $BUILD_DIR
+rm -rf mingw-w64-v10.0.0
diff -Naur cross-sh/toolchain_musl.sh cross-sh-mingw/toolchain_musl.sh
--- cross-sh/toolchain_musl.sh	2022-09-18 13:08:24.964526719 -0400
+++ cross-sh-mingw/toolchain_musl.sh	1969-12-31 19:00:00.000000000 -0500
@@ -1,26 +0,0 @@
-#!/bin/bash
-set -e
-# Source the configuration
-source toolchain.conf
-
-pkg=musl
-# Untar
-
-cd $BUILD_DIR
-tar -xvf $SOURCES_DIR/$pkg-$MUSL_VER.tar.gz
-cd $pkg-$MUSL_VER
-
-# Start the build
-
-mkdir build
-cd build
-
-../configure --prefix=/usr --host=$TARGET_CROSS 
-
-make -j$JOBS
-
-make DESTDIR=$LOCATION install
-
-# Delete the old files
-cd $BUILD_DIR
-rm -rf $pkg-$MUSL_VER
diff -Naur cross-sh/toolchain_newlib.sh cross-sh-mingw/toolchain_newlib.sh
--- cross-sh/toolchain_newlib.sh	2022-09-18 13:08:24.964526719 -0400
+++ cross-sh-mingw/toolchain_newlib.sh	1969-12-31 19:00:00.000000000 -0500
@@ -1,26 +0,0 @@
-#!/bin/bash
-set -e
-# Source the configuration
-source toolchain.conf
-
-pkg=newlib
-# Untar
-
-cd $BUILD_DIR
-tar -xvf $SOURCES_DIR/$pkg-$NEWLIB_VER.tar.xz
-cd $pkg-$NEWLIB_VER
-
-# Start the build
-
-mkdir build
-cd build
-
-../configure --prefix=/usr --host=$TARGET_CROSS --with-headers=$LOCATION/include
-
-make -j$JOBS
-
-make DESTDIR=$LOCATION install
-
-# Delete the old files
-cd $BUILD_DIR
-rm -rf $pkg-$NEWLIB_VER
diff -Naur cross-sh/toolchain_picolibc.sh cross-sh-mingw/toolchain_picolibc.sh
--- cross-sh/toolchain_picolibc.sh	2022-09-18 13:08:24.964526719 -0400
+++ cross-sh-mingw/toolchain_picolibc.sh	1969-12-31 19:00:00.000000000 -0500
@@ -1,26 +0,0 @@
-#!/bin/bash
-set -e
-# Source the configuration
-source toolchain.conf
-
-pkg=picolibc
-# Untar
-
-cd $BUILD_DIR
-tar -xvf $SOURCES_DIR/$pkg-$PICOLIBC_VER.tar.xz
-cd $pkg-$PICOLIBC_VER
-
-# Start the build
-
-mkdir build
-cd build
-
-../configure --prefix=$LOCATION/usr --host=$TARGET_CROSS --with-headers=$LOCATION/include
-
-make -j$JOBS
-
-make install
-
-# Delete the old files
-cd $BUILD_DIR
-rm -rf $pkg-$PICOLIBC_VER
diff -Naur cross-sh/wget cross-sh-mingw/wget
--- cross-sh/wget	2022-09-18 13:08:24.965526723 -0400
+++ cross-sh-mingw/wget	2022-09-18 13:06:51.134172663 -0400
@@ -1,10 +1,3 @@
-https://ftp.gnu.org/gnu/gcc/gcc-12.1.0/gcc-12.1.0.tar.xz
-https://cdn.kernel.org/pub/linux/kernel/v5.x/linux-5.18.10.tar.xz
-https://ftp.gnu.org/gnu/binutils/binutils-2.38.tar.xz
-https://ftp.gnu.org/gnu/glibc/glibc-2.35.tar.xz
-https://ftp.gnu.org/gnu/gmp/gmp-6.2.1.tar.xz
-https://ftp.gnu.org/gnu/mpc/mpc-1.2.1.tar.gz
-https://ftp.gnu.org/gnu/mpfr/mpfr-4.1.0.tar.xz
-https://musl.libc.org/releases/musl-1.2.3.tar.gz
-ftp://sourceware.org/pub/newlib/newlib-4.2.0.20211231.tar.gz
-https://github.com/picolibc/picolibc/releases/download/1.7.7/picolibc-1.7.7.tar.xz
+https://ftp.gnu.org/gnu/gcc/gcc-12.2.0/gcc-12.2.0.tar.xz
+https://ftp.gnu.org/gnu/binutils/binutils-2.39.tar.xz
+https://sourceforge.net/projects/mingw-w64/files/mingw-w64/mingw-w64-release/mingw-w64-v10.0.0.tar.bz2
